rdrf/rdrf/views/patient_view.py

```diff
@@ -588,7 +590,7 @@ def _blacklisted_sections(self, registry_model):
         return [_(x) for x in registry_model.metadata.get('section_blacklist', [])]
 
 
-class AddPatientView(PatientFormMixin, CreateView):
+class AddPatientView(StaffMemberRequiredMixin, PatientFormMixin, CreateView):
```

Comment:
Again, seems too lax.
Compare to the checks we do on a user when they want to edit a patient:  `security_check_user_patient`.

We should have similar checks when adding a patient. The patient object doesn't exist of course, but still the rules should be similar. Allow adding only if superuser or special user (curator, clinician etc.) and the registry matches.

Reply:
Updated.